### ISSUE TITLE
Static constructors of obfuscated classes produce a crash

### ISSUE THREAD

**Issue by Ryhon0** [BepInEx/BepInEx#658]
### What happened?

Triggering a static constructor of an obfuscated class referenced in BattleBit Remastered produces a crash.
The game is using IL2CPP and looks to be obfuscated using Beebyte Obfuscator.

### BepInEx Distribution

Bleeding Edge from BepisBuilds

### Log outputs

```txt
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
Repeat 2 times:
--------------------------------
   at Il2CppInterop.Runtime.IL2CPP.il2cpp_runtime_class_init(IntPtr)
--------------------------------
   at MonoBehaviourPublicCaGaCaReUICaGa1ObLiUnique..cctor()
   at System.RuntimeFieldHandle.GetValue(System.Reflection.RtFieldInfo, System.Object, System.RuntimeType, System.RuntimeType, Boolean ByRef)
   at System.Reflection.RtFieldInfo.GetValue(System.Object)
   at Il2CppInterop.HarmonySupport.Il2CppDetourMethodPatcher.Init()
   at Il2CppInterop.HarmonySupport.Il2CppDetourMethodPatcher..ctor(System.Reflection.MethodBase)
   at Il2CppInterop.HarmonySupport.HarmonySupportComponent.TryResolve(System.Object, PatcherResolverEventArgs)
   at HarmonyLib.Public.Patching.PatchManager.GetMethodPatcher(System.Reflection.MethodBase)
   at HarmonyLib.PatchFunctions.UpdateWrapper(System.Reflection.MethodBase, HarmonyLib.PatchInfo)
   at HarmonyLib.PatchClassProcessor.ProcessPatchJob(Job<System.Reflection.MethodInfo>)
   at HarmonyLib.PatchClassProcessor.PatchWithAttributes(System.Reflection.MethodBase ByRef)
   at HarmonyLib.PatchClassProcessor.Patch()
   at HarmonyLib.Harmony.<PatchAll>b__11_0(System.Type)
   at HarmonyLib.CollectionExtensions.Do[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.Collections.Generic.IEnumerable`1<System.__Canon>, System.Action`1<System.__Canon>)
   at HarmonyLib.Harmony.PatchAll(System.Reflection.Assembly)
   at HarmonyLib.Harmony.PatchAll()
   at BattleMod.Plugin.Load()
   at BepInEx.Unity.IL2CPP.IL2CPPChainloader.LoadPlugin(BepInEx.PluginInfo, System.Reflection.Assembly)
   at BepInEx.Bootstrap.BaseChainloader`1[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].LoadPlugins(System.Collections.Generic.IList`1<BepInEx.PluginInfo>)
   at BepInEx.Bootstrap.BaseChainloader`1[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].Execute()
   at BepInEx.Unity.IL2CPP.IL2CPPChainloader.OnInvokeMethod(IntPtr, IntPtr, IntPtr, IntPtr)
```

### Environment

```markdown
- OS: Arch Linux
- BepInEx: 6.0.0-be.672
- Game: BattleBit Remastered
```

**Comment by unreliablecode** [comment]
i think this one should fix it [Il2CppInterop pull #101](https://github.com/BepInEx/Il2CppInterop/pull/101)

**Comment by Ryhon0** [comment]
Works perfectly, thank you!
